packages/core/src/ctype/CType.chain.ts

```diff
@@ -33,12 +46,166 @@ export function idToChain(cTypeId: ICType['$id']): CTypeHash {
   return idToHash(cTypeId)
 }
 
+// Transform a blockchain-formatted CType input (represented as Bytes) into the original [[IPublicCredentialInput]].
+// It throws if what was written on the chain was garbage.
+function cTypeInputFromChain(input: Bytes): ICType {
+  try {
+    // Throws on invalid JSON input. CType is expected to be a valid JSON document.
+    const reconstructedObject = JSON.parse(input.toUtf8())
+    // Re-compute the ID to validate the resulting ICType.
+    const reconstructedCTypeId = hashToId(getHashForSchema(reconstructedObject))
+    const reconstructedCType: ICType = {
+      ...reconstructedObject,
+      $id: reconstructedCTypeId,
+    }
+    // If throws if the input was a valid JSON but not a valid CType.
+    verifyDataStructure(reconstructedCType)
+    return reconstructedCType
+  } catch (cause) {
+    throw new SDKErrors.CTypeError(
+      `The provided payload cannot be parsed as a CType: ${input.toHuman()}`,
+      { cause }
+    )
+  }
+}
+
+/**
+ * The details of a CType that are stored on chain.
+ */
+export interface CTypeChainDetails {
+  /**
+   * The DID of the CType's creator.
+   */
+  creator: DidUri
+  /**
+   * The block number in which the CType was created.
+   */
+  createdAt: BN
+}
+
```

Comment:
also to be replaced with autogenerated types

Reply:
Same as above.